### ISSUE TITLE
Ivy and rgrep from a dired buffer

### ISSUE THREAD

**Issue by pasja** [abo-abo/swiper#676]
Hello abo-abo,

Fist and foremost, thanks for this wonderful package! 

On the other hand i am sad that i bring up another issue with rgrep and ivy, but i think i have found a corner case what `ivy-immediate-done` does not cover, namely if i start an `rgrep` session from a `dired` buffer, than it puts the directory name into the pattern.

I have looked into the source code and make a modified version for myself. here is the diff of the function:
```diff
 (defun ivy-immediate-done ()
   "Exit the minibuffer with current input instead of current candidate."
   (interactive)
   (delete-minibuffer-contents)
-  (insert (setq ivy--current
-                (if ivy--directory
-                    (expand-file-name ivy-text ivy--directory)
-                  ivy-text)))
+  (insert (setq ivy--current ivy-text))
   (setq ivy-exit 'done)
   (exit-minibuffer))
```

What do you think, could this patch be merged to ivy mainline?

**Comment by abo-abo** [comment]
> namely if i start an rgrep session from a dired buffer, than it puts the directory name into the pattern.

Can't reproduce, please give more details.

> What do you think, could this patch be merged to ivy mainline?

The code that you remove was put there for a reason. Your change would break the code in a different place.

**Comment by pasja** [comment]
Hello abo-abo,

I have made another with a clean emacs environment `emacs -Q` and loaded only `ivy.el` and `colir.el`.
After that opened a dired buffer and started an rgrep in that buffer.

I have noticed that a new option appeared, namely `all`. With that rgrep worked as i expected from them.
I went back to my original config, and for my amusement, `all` appeared there as well.

So there is no problem with ivy, only i was a little bit overhasty with that report. 
Thanks four your quick answer, with that i close this issue.
